### ISSUE TITLE
Expose default menu to allow it to be tweaked in a custom renderMenu

### ISSUE THREAD

**Issue by alasdairhurst** [ericgio/react-bootstrap-typeahead#312]
Currently, if you are specifying a custom menu you need to re-implement the whole thing even if you just want to make a small change.

Example:
I want to control the visibility of the menu based on the current value of the input field, and the available results.

Implementation:
if there is a certain menuProps.text value, dont render anything
otherwise return the default menu with a custom classname

From what I see, i have to import  Highlighter, Menu and MenuItem, then implement renderMenuItemChildren with highlighter:
```
renderMenuItemChildren: (option, props) => {
		return (
			<Highlighter search={props.text}>
				{option}
			</Highlighter>
		);
	},
```

And then re-implement the menu renderer
```
	return (
			<Menu {...menuProps} className="typeahead typeahead-menu">
				{results.map((result, index) => (
					<MenuItem option={result} position={index} key={`${result}_${index}`}>
						{renderMenuItemChildren(result, menuProps)}
					</MenuItem>
				))}
			</Menu>
		);
```

Ideally i would just like to be able to do something like this, without having to re-implement all the rendering.
```
return Typeahead.defaultMenu(results, { ...menuProps, className: 'typeahead typeahead-menu' });
```
or

```
 return <Typeahead.DefaultMenu results={results} className="typeahead typeahead-menu"
 {...menuProps} />
```

Basically exposing the component you use here:
https://github.com/ericgio/react-bootstrap-typeahead/blob/bf6ea72cb6a13dd0f75480490f64eb4c6dc5491d/src/Typeahead.react.js#L95

**Comment by ericgio** [comment]
It's easy enough to expose `TypeaheadMenu` as a top-level component, but it seems like what you really want is the ability to control the menu's visibility and this is just a workaround. I'm going to close this issue in favor of #311, but feel free to reopen if you feel I'm misunderstanding something.

**Comment by alasdairhurst** [comment]
yes and no. The main functionality is to hide the menu, but I also want to set a class on the menu to scope any style changes that I make to my wrapper component.

**Comment by ericgio** [comment]
Note that while not exposed at the top level, you should be able to do:

```js
import TypeaheadMenu from 'react-bootstrap-typeahead/lib/TypeaheadMenu';
```
